### PR TITLE
Add basic Doctrine TestSuite and Character Model

### PR DESCRIPTION
**Pull request by Vassyli**
Adds a basic Doctrine TestSuite which provides scheme creation and supports default data using the yaml format.

Also adds some basic Exceptions as well as a basic Character Model.

### CODE REVIEW

**Comment by Vassyli** [comment]
Why are my tests always failing with travis, but are passing locally?

**Comment by austenmc** [comment]
You can see why it fails by clicking through to Details in the "All checks have failed" box.

What happened was this: the `t` script is just running `phpunit` with no arguments, which will read out the config from the xml file. But in this mode, it expects some testsuites to be defined in the xml file. 

Since you want to preserve the config info in the xml file, I would add a generic testsuite:

```
    <testsuites>
        <testsuite name="All">
            <directory>tests/</directory>
        </testsuite>
    </testsuites>
```

File: src/Models/Character.php [hunk 1]
```diff
@@ -46,4 +65,46 @@ public function getName(): string {
...
+    }
+    
+    /**
+     * Sets the maximum health of a character to a given value. It also sets the
+     * health if none has been set yet.
+     * @param int $maxhealth
+     */
+    public function setMaxHealth(int $maxHealth) {
+        $this->maxHealth = $maxHealth;
+        $this->health = $this->health ?? $this->maxHealth;
```
Comment (austenmc) [inline src/Models/Character.php:84]: I still feel like this shouldnt be here as an unexpected side effect.

File: src/Tools/Model/Creator.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                
+                $entity->$methodname($value);
+                unset($arguments[$field]);
+            }
+        }
+        
+        if(count($arguments) > 0) {
+            throw new UnexpectedArrayKeyException('self::$fillable does allow the properties "'.implode(", ", array_keys($arguments)).'" to be set.');
+        }
+
```
Comment (austenmc) [inline src/Tools/Model/Creator.php:49]: thumbs-up

File: tests/ModelTestCase.php [hunk 3]
```diff
@@ -1,46 +1,54 @@
...
-    /**
-     * Sets up the database and database structure
-     */
-    protected function setUp() {        
-        $configuration = Setup::createAnnotationMetadataConfiguration(["src/Models"], true);
-        $configuration->setQuoteStrategy(new AnsiQuoteStrategy());
+    final public function getConnection() {
+        if($this->connection === null) {
+            if(self::$pdo === null) {
+                self::$pdo = new \PDO($GLOBALS['DB_DSN'], $GLOBALS["DB_USER"], $GLOBALS["DB_PASSWD"]);
```
Comment (austenmc) [inline tests/ModelTestCase.php:25]: I know this is small, but any reason not to say `DB_PASSWORD`? Why abbreviate?

**Pull request closed without merging**